internal/zinc-persist/src/main/scala/sbt/internal/inc/mappers/MapperUtils.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Zinc - The incremental compiler for Scala.
+ * Copyright 2011 - 2017, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * This software is released under the terms written in LICENSE.
+ */
+
+package sbt.internal.inc.mappers
+
+import java.io.File
+import java.nio.file.Path
+
+import xsbti.compile.analysis.Stamp
+
+object MapperUtils {
+  private[inc] def rebase(target: File, from: Path, to: Path): File = {
+    to.resolve(from.relativize(target.toPath)).toFile
+  }
+
+  /** Defines a marker that tells the utils that the relativized path is empty. */
+  private final val RELATIVE_MARKER = "\u2603\u2603\u2603"
```

Comment:
This is the unicode symbol used internally for the synthesized macro names by sbt. It's safe to be used because the text format uses UTF8 (we force it when we write and read) and protobuf also uses unicode (https://developers.google.com/protocol-buffers/docs/proto, look for "unicode" and "UTF").
  
Given the fact that this symbol is rarely used, I think it's perfect for this scenario. Some SO questions point to it as a good way to disambiguate names or use it for "marking" string boundaries.